Libplanet.Explorer/Store/MySQLRichStore.cs

```diff
@@ -256,19 +309,162 @@ public Block<T> GetBlock<T>(HashDigest<SHA256> blockHash)
+                {
+                    var query = db.Query(BlockDbName).Where("miner", miner.ToString())
+                        .Offset(offset)
+                        .Limit((int)limit)
+                        .Select("hash");
+                    query = desc ? query.OrderByDesc("index") : query.OrderBy("index");
+                    return query.OrderBy("index")
+                        .Get<string>()
+                        .Select(hashString => new HashDigest<SHA256>(
+                            ByteUtil.ParseHex(hashString)));
+                }
+                else
+                {
+                    var query = db.Query(BlockDbName).Where("miner", miner.ToString())
+                        .Offset(offset)
+                        .Select("hash");
+                    query = desc ? query.OrderByDesc("index") : query.OrderBy("index");
+                    return query.OrderBy("index")
+                        .Get<string>()
+                        .Select(hashString => new HashDigest<SHA256>(
+                            ByteUtil.ParseHex(hashString)));
+                }
+            }
+            else
+            {
+                if (limit != null)
+                {
+                    var query = desc ? db.Query(BlockDbName).OrderByDesc("index") :
+                        db.Query(BlockDbName).OrderBy("index");
+                    query = query.Offset(offset).Select("hash");
+                    return query.OrderBy("index")
+                        .Get<string>()
+                        .Select(hashString => new HashDigest<SHA256>(
+                            ByteUtil.ParseHex(hashString)));
+                }
+                else
+                {
+                    var query = desc ? db.Query(BlockDbName).OrderByDesc("index") :
+                        db.Query(BlockDbName).OrderBy("index");
+                    query = query.Offset(offset).Limit((int)limit).Select("hash");
+                    return query.OrderBy("index")
+                        .Get<string>()
+                        .Select(hashString => new HashDigest<SHA256>(
+                            ByteUtil.ParseHex(hashString)));
+                }
+            }
+        }
+
+        public IEnumerable<HashDigest<SHA256>> GetBlockHashesWithTx(
+            bool desc,
+            int offset,
+            int? limit,
+            Address? miner)
+        {
+            using QueryFactory db = OpenDB();
+            if (!(miner is null))
+            {
+                if (limit != null)
+                {
+                    var query = db.Query(BlockDbName).Where("miner", miner.ToString())
+                        .WhereNot("tx_hash", string.Empty)
+                        .Offset(offset)
+                        .Limit((int)limit)
+                        .Select("hash");
+                    query = desc ? query.OrderByDesc("index") : query.OrderBy("index");
+                    return query.OrderBy("index")
```

Comment:
It's hard to understand these lines to me. `query` seems already sorted by `desc` parameter but it seems to order by ascending in force. Is it intended?

Reply:
@moreal
You're right, I think it's related to @ipdae's [comment](https://github.com/planetarium/libplanet-explorer/pull/165/files#r594969205). I'll make corrections.